cmd/registry/tests/task-validation_test.go

```diff
@@ -0,0 +1,147 @@
+			want:   "compute",
+		},
+		{
+			desc:   "delete action",
+			action: "delete projects/demo/apis/-/versions/-/specs/-",
+			want:   "delete",
+		},
+		{
+			desc:   "export action",
+			action: "export csv projects/demo/apis/petstore/versions/1.0.0",
+			want:   "export",
+		},
+		{
+			desc:   "get action",
+			action: "get projects/demo/apis/petstore/versions/1.0.0/specs/openapi.yaml",
+			want:   "get",
+		},
+		{
+			// Index is currently experimental
+			// This is a placeholder test case not necessarily the right usage
+			desc:   "index action",
+			action: "index projects/demo/apis/petstore/versions/1.0.0/specs/openapi.yaml",
+			want:   "index",
+		},
+		{
+			desc:   "label action",
+			action: "label projects/demo/apis/petstore/versions/1.0.0/specs/openapi.yaml env=test --overwrite",
+			want:   "label",
+		},
+		{
+			desc:   "list action",
+			action: "list projects/demo/apis/-/versions/-/specs/-",
+			want:   "list",
+		},
+		{
+			// Search is currently experimental
+			// This is a placeholder test case not necessarily the right usage
+			desc:   "search action",
+			action: "search 'API'",
+			want:   "search",
+		},
+		{
+			desc:   "upload action",
+			action: "upload manifest cmd/registry/controller/testdata/manifest_e2e.yaml --project_id=demo",
+			want:   "upload",
+		},
+		{
+			desc:   "vocabulary action",
+			action: "vocabulary union projects/demo/apis/petstore/versions/2.0.0/specs/openapi.yaml/artifacts/vocabulary projects/demo/apis/petstore/versions/5.0.0/specs/openapi.yaml/artifacts/vocabulary",
+			want:   "vocabulary",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			ctx := context.Background()
+			cmd, err := controller.GetCommand(ctx, strings.Fields(test.action))
+			if err != nil {
+				t.Fatalf("Error executing GetCommand() %s", err)
+			}
+
+			if cmd.Name() != test.want {
+				t.Errorf("GetCommand() created unexpected Command instance. Want: %s, got: %s", test.want, cmd.Name())
+			}
+		})
+	}
+}
+
+// Test the error scenarios for GetCommand()
+func TestErrorCases(t *testing.T) {
+	tests := []struct {
+		desc   string
+		action string
+		want   string
+	}{
+		{
+			desc:   "resolve action",
+			action: "resolve projects/demo/artifacts/test-manifest",
+			want:   "Cannot use command 'resolve' in action.",
+		},
+		{
+			desc:   "invalid action",
+			action: "random projects/demo/apis/petstore",
+			want:   "Action 'random projects/demo/apis/petstore' is not supported.",
+		},
+		{
+			desc:   "empty action",
+			action: "",
+			want:   "Empty action string.",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			ctx := context.Background()
+			_, err := controller.GetCommand(ctx, strings.Fields(test.action))
+			if err == nil {
+				t.Errorf("Expected GetCommand() to return error.")
+			}
+			if err.Error() != test.want {
+				t.Errorf("Generated unexpected error message. Want: %s, got: %s", test.want, err)
+			}
```

Comment:
I don't think we should be testing the error message specifically because changing the error message isn't something we want to prevent. Here are two suggestions:

1. Remove this check so we only fail if `err == nil`. I think this is good enough, but it's possible a test case could incorrectly pass when a different error than we expect occurs. 
2. Export specific errors as variables so we can check for specific errors without relying on their message. For example, see the [os package](https://pkg.go.dev/os#pkg-variables). Packages usually only do this if the callers will handle the errors differently, e.g. ignoring not found errors and returning others.

Reply:
Makes sense. I am not exporting the errors because the client is not treating any of the messages differently. The error message is just for debug purposes.